packages/frontend/src/components/shifts/ShiftDisplay.tsx

```diff
@@ -39,6 +64,12 @@ export default function ShiftDisplay() {
     handleDayChange(-1);
   };
 
+  useEffect(() => {
+    scheduleClient.GetAllSchedules().then((loadedSchedules) => {
```

Comment:
1. Personally I hate `then` syntax but might just be me
2. You probably don't want this effect running on every render

```js
useEffect(() => {
  (async () => {
    const loadedSchedules = await scheduleClient.GetAllSchedules();
    setSchedules(loadedSchedules);
  })();
}, [scheduleClient]);
```

Reply:
Yeah i agree. await is much cleaner.

Reply:
I made this effect depend on the `currentDay`. This should reduce its execution to only when the "switch day" btns are clicked.